doc/api/punycode.md

```diff
@@ -1,6 +1,11 @@
 # punycode
 
-    Stability: 2 - Stable
+    Stability: 0 - Deprecated
+
+**The version of the punycode module bundled in Node.js is being deprecated**.
+In a future major version of Node.js this module will be removed. Users
```

Comment:
@jsnell as we have added this pull-request to the 7.0.0 milestone, so could we just simply make it be deprecated entirely at v8 as we will notify user at v7.